src/loaders/jobs/taskfarmer/task_generator.py

```diff
@@ -0,0 +1,284 @@
+    """
+
+    # The content of the Shifter wrapper script
+    shifter_wrapper = '''#!/bin/bash\n\n'''
+    shifter_wrapper += f'''image={image_str}\n\n'''
+    shifter_wrapper += '''if [ $# -lt 1 ]; then\n'''
+    shifter_wrapper += '''    echo "Error: Missing command argument."\n'''
+    shifter_wrapper += '''    echo "Usage: shifter_wrapper.sh your-command-arguments"\n'''
+    shifter_wrapper += '''    exit 1\n'''
+    shifter_wrapper += '''fi\n\n'''
+    shifter_wrapper += '''command=\"$@\"\n\n'''
+    shifter_wrapper += '''echo \"Running shifter --image=$image $command\"\n\n'''
+    shifter_wrapper += f'''cd {job_dir}\n'''
+    shifter_wrapper += '''shifter --image=$image $command\n'''  # Run the command in the Shifter environment
+
+    wrapper_file = os.path.join(job_dir, 'shifter_wrapper.sh')
+    with open(wrapper_file, "w") as f:
+        f.write(shifter_wrapper)
+
+    os.chmod(wrapper_file, 0o777)
+
+    return wrapper_file
+
+
+def _create_genome_id_file(genome_ids, genome_id_file):
+    """
+    Create a tab-separated values (TSV) file with a list of genome IDs.
+    """
+    with open(genome_id_file, 'w') as f:
+        f.write("genome_id\n")
+        for genome_id in genome_ids:
+            f.write(f"{genome_id}\n")
+
+
+def _create_task_list(source_data_dir, kbase_collection, load_ver, tool, wrapper_file, job_dir, root_dir):
+    """
+    Create task list file (tasks.txt)
+    """
+    genome_ids = [path for path in os.listdir(source_data_dir) if
+                  os.path.isdir(os.path.join(source_data_dir, path))]
+
+    chunk_size = CHUNK_SIZE[tool]
+    genome_ids_chunks = [genome_ids[i: i + chunk_size] for i in range(0, len(genome_ids), chunk_size)]
+
+    task_list = '#!/usr/bin/env bash\n'
+    for idx, genome_ids_chunk in enumerate(genome_ids_chunks):
+
+        task_list += wrapper_file
+
+        genome_id_file = os.path.join(job_dir, f'genome_id_{idx}.tsv')
+        _create_genome_id_file(genome_ids_chunk, genome_id_file)
+
+        if tool == 'checkm2':
+            task_list += f'	--volume={CHECKM2_DB}:/CheckM2_database ' \
+                         f'conda run -n checkm2-1.0.0 ' \
+                         f'python collections/src/loaders/genome_collection/compute_genome_attribs.py ' \
+                         f'--tools checkm2 '
+        elif tool == 'gtdb_tk':
+            task_list += f' --volume={GTDBTK_DATA_PATH}:/gtdbtk_reference_data ' \
+                         f'conda run -n gtdbtk-2.1.1 ' \
+                         f'python collections/src/loaders/genome_collection/compute_genome_attribs.py ' \
+                         f'--tools gtdb_tk '
+        else:
+            raise ValueError(f'Unexpected tool {tool}')
+
+        task_list += f'--load_ver {load_ver} --source_data_dir {source_data_dir} ' \
+                     f'--kbase_collection {kbase_collection} --root_dir {root_dir} ' \
+                     f'--threads 128 --program_threads 128 --node_id job_{idx} ' \
```

Comment:
using `/` line continuations is discouraged by pep8

Reply:
👍

Reply:
:+1: